protocol_grpc.go

```diff
@@ -319,6 +315,51 @@ func (g *grpcClient) NewConn(
 	return wrapClientConnWithCodedErrors(conn)
 }
 
+func (g *grpcClient) Invoke(
+	ctx context.Context, spec Spec, request AnyRequest, response AnyResponse,
+) error {
+	// TODO: should we have a grpcUnaryClient struct?
```

Comment:
@jhump how did you see this Invoke implementation working? I started to flesh out what is needed, but feels like it's just copy pasta of the streaming. Shall I factor out all the shareable parts of the streaming call?